doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -91,47 +206,72 @@ In this case, no spaces are allowed in the symbol.  Also, if the
+`"1 + 2 * 2"` is parsed as `1 + (2 * 2)` rather than `(1 + 2) *
+2`. Left associativity mandates that `x + y + z` gets parsed as
+`(x + y) + z` rather than `x + (y + z)`. More details can be found in
+the :ref:`precedences and associativity <PrecedencesAndAssociativity>`
+section below.
 
-   Occurs when the notation string contains an unterminated quoted
-   string, as e.g. in :g:`Reserved Notation "A ""an unended string B"`, for which the
-   user may instead mean :g:`Reserved Notation "A ""an ended string"" B`.
+.. note:: The notations mentioned in the module :ref:`init-notations` are already reserved. Hence
+          their precedence and associativity cannot be changed.
 
-.. exn:: End of quoted string not followed by a space in notation.
+The :cmd:`Reserved Notation` command not only modifies the parser,
+it can also modify the lexer.
 
-   Occurs when the notation string contains a quoted string which
-   contains a double quote not ending the quoted string, as e.g. in
-   :g:`Reserved Notation "A ""string""! B"` or `Reserved Notation "A ""string""!"" B"`, for which
-   the user may instead mean :g:`Reserved Notation "A ""string"""" ! B`,
-   :g:`Reserved Notation "A ""string""""!"" B`, or :g:`Reserved Notation "A '""string""!' B`.
+.. coqtop:: in
+
+   Reserved Notation "'+2'".
+
+This modifies the lexer, adding the keyword `'+2'`. Now, if the string
+of the previous example is written without spaces, i.e., ``"1+2*2"``,
+it would be lexed as four tokens `'1' '+2' '*' '2''` instead of the
+previous five. This example shows that new keywords should be
+introduced with a lot of care.
+
+.. cmd:: Reserved Infix @string {? ( {+, @syntax_modifier } ) }
+
+   This command declares an infix :term:`parsing` rule.
+
+When a format is attached to a reserved notation (with the `format`
+:token:`syntax_modifier`), it is used by default for printing all
+subsequent interpretations of the corresponding
+notation. Individual interpretations can override the format.
+
+.. warn:: Notations "a b" defined at level x and "a c" defined at level y have incompatible prefixes. One of them will likely not work.
+   :name: notation-incompatible-prefix
+
+   The two notations have a common prefix but different levels.
+   The levels of one of the notations should be adjusted to match
+   the other. See :ref:`factorization <NotationFactorization>` for
+   details.
+
+.. _PrecedencesAndAssociativity:
 
 Precedences and associativity
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Mixing different symbolic notations in the same text may cause serious
-parsing ambiguity. To deal with the ambiguity of notations, Coq uses
+:term:`parsing` ambiguity. To deal with the ambiguity of notations, Coq uses
 precedence levels ranging from 0 to 100 (plus one extra level numbered
 200) and associativity rules.
 
 Consider for example the new notation
 
 .. coqtop:: in
 
-   Notation "A \/ B" := (or A B).
+   Reserved Notation "A \/ B".
```

Comment:
What's the rationale for changing all the examples to use `Reserved Notation`?  I don't see the point.  Elsewhere in the chapter the doc can explain when it's best practice or required to use reservations.  It could also say in one place that the examples often use just `Notation` for simplicity.

I think it's helpful for readers to see what the notation is transformed into (ie `(or A B)`).

Reply:
Agree, c.f. above comment.